storm-core/src/clj/backtype/storm/command/blobstore.clj

```diff
@@ -0,0 +1,163 @@
+  "Create a blob in the blob store from an InputStream"
+  [key ^InputStream in ^SettableBlobMeta meta]
+  (with-configured-blob-client blobstore
+    (let [out (.createBlob blobstore key meta)]
+      (try 
+        (copy in out)
+        (.close out)
+        (catch Exception e
+          (.cancel out)
+          (throw e))))))
+
+(defn read-blob
+  "Read a blob in the blob store and write to an OutputStream"
+  [key ^OutputStream out]
+  (with-configured-blob-client blobstore
+    (with-open [in (.getBlob blobstore key)]
+      (copy in out))))
+
+(defn as-access-control
+  "Convert a parameter to an AccessControl object"
+  [param]
+  (BlobStoreAclHandler/parseAccessControl (str param)))
+
+(defn as-acl
+  [param]
+  (map as-access-control (split param #",")))
+
+(defn access-control-str
+  [^AccessControl acl]
+  (BlobStoreAclHandler/accessControlToString acl))
+
+(defn read-cli [args]
+  (let [[{file :file} [key] _] (cli args ["-f" "--file" :default nil])]
+    (if file
+      (with-open [f (output-stream file)]
+        (read-blob key f))
+      (read-blob key System/out))))
+
+(defn update-cli [args]
+  (let [[{file :file} [key] _] (cli args ["-f" "--file" :default nil])]
+    (if file
+      (with-open [f (input-stream file)]
+        (update-blob-from-stream key f))
+      (update-blob-from-stream key System/in))
+    (log-message "Successfully updated " key)))
+
+(defn create-cli [args]
+  (let [[{file :file acl :acl replication-factor :replication-fctr} [key] _] (cli args ["-f" "--file" :default nil]
+                                                  ["-a" "--acl" :default [] :parse-fn as-acl]
+                                                  ["-r" "--replication-factor" :default -1 :parse-fn parse-int])
+        meta (doto (SettableBlobMeta. acl)
+                   (.set_replication_factor replication-factor))]
+    (log-message "Creating " key " with ACL " (pr-str (map access-control-str acl)))
+    (if file
+      (with-open [f (input-stream file)]
+        (create-blob-from-stream key f meta))
+      (create-blob-from-stream key System/in meta))
+    (log-message "Successfully created " key)))
+
+(defn delete-cli [args]
+  (with-configured-blob-client blobstore
+    (doseq [key args]
+      (.deleteBlob blobstore key)
+      (log-message "deleted " key))))
+
+(defn list-cli [args]
+  (with-configured-blob-client blobstore
+    (let [keys (if (empty? args) (iterator-seq (.listKeys blobstore)) args)]
+      (doseq [key keys]
+        (try
+          (let [meta (.getBlobMeta blobstore key)
+                version (.get_version meta)
+                acl (.get_acl (.get_settable meta))]
+            (log-message key " " version " " (pr-str (map access-control-str acl))))
+          (catch AuthorizationException ae
+            (if-not (empty? args) (log-message "ACCESS DENIED to key: " key)))
+          (catch KeyNotFoundException knf
+            (if-not (empty? args) (log-message key " NOT FOUND"))))))))
```

Comment:
It would be nice to log the two messages inside the `catch` forms to STDERR instead of STDOUT.